railties/test/generators/app_generator_test.rb

```diff
@@ -1110,6 +1109,38 @@ def test_master_key_is_only_readable_by_the_owner
     end
   end
 
+  def test_minimal_rails_app
+    app_root = File.join(destination_root, "myapp")
+    run_generator [app_root, "--minimal"]
+
+    stub_rails_application(app_root) do
+      generator = Rails::Generators::AppGenerator.new ["rails"], { skip_webpack_install: true,
+        skip_javascript: true, skip_active_storage: true, skip_action_cable: true, skip_bootsnap: true,
+        skip_turbolinks: true, skip_spring: true, skip_action_text: true, skip_action_mailer: true,
+        skip_action_mailbox: true }, { destination_root: app_root, shell: @shell }
+      generator.send(:app_const)
+      quietly { generator.send(:update_config_files) }
```

Comment:
I believe you can remove this section.  This boilerplate is for testing `app:update`.  Since `--minimal` is a kind of meta option, I think we can leave testing `app:update` to each individual option.

Reply:
Sounds good, we did not have a test for --api so was not too sure about this.

I have updated the test, I think this is what you meant, let me know if that's not the case:

```
def test_minimal_rails_app
    app_root = File.join(destination_root, "myapp")
    run_generator [app_root, "--minimal"]

    assert_no_file "#{app_root}/config/storage.yml"
    assert_no_file "#{app_root}/config/webpacker.yml"
    assert_no_file "#{app_root}/config/cable.yml"
    assert_no_file "#{app_root}/bin/yarn"
    assert_no_file "#{app_root}/views/layouts/mailer.html.erb"
    assert_no_file "#{app_root}/config/spring.rb"
    assert_file "#{app_root}/app/views/layouts/application.html.erb" do |content|
      assert_no_match(/data-turbolinks-track/, content)
    end
    assert_file "#{app_root}/config/environments/development.rb" do |content|
      assert_no_match(/config\.active_storage/, content)
    end
    assert_file "#{app_root}/config/boot.rb" do |content|
      assert_no_match(/require "bootsnap\/setup"/, content)
    end
    assert_file "#{app_root}/config/application.rb", /#\s+require\s+["']action_text\/engine["']/
    assert_file "#{app_root}/config/application.rb", /#\s+require\s+["']action_mailbox\/engine["']/
  end
```

Reply:
Yup, that's what I meant! :+1: